backend/x86_proc.ml

```diff
@@ -285,19 +349,39 @@ let assemble_file infile outfile =
   | Some content -> content outfile; binary_content := None; 0
 
 let asm_code = ref []
+let asm_code_by_sec = ref (ref [])
+let asm_sections = SectionTbl.create 100
```

Comment:
The naming is confusing here.  I would rename `asm_code_by_sec -> asm_code_current_section` and `asm_sections -> asm_code_by_section`.

Reply:
asm_code_by_sec was intended to be the hash table, im not sure how it became that